### PR TITLE
[10.0][FIX] account invoice triple discount: onchange cache management

### PR DESCRIPTION
**Pull request by SimoRubi**
Fix https://github.com/OCA/sale-workflow/issues/817

### CODE REVIEW

File: account_invoice_triple_discount/models/account_invoice.py [hunk 1]
```diff
@@ -54,6 +54,9 @@ def triple_discount_preprocess(self):
         Resetting discount2 and discount3 to 0.0 avoids issues if
         this method is called multiple times."""
         prev_values = dict()
+        if not self.env.in_onchange:
```
Comment (lmignon) [inline account_invoice_triple_discount/models/account_invoice.py:57]: @SimoRubi Can you elaborate why this trick is required?

Comment (SimoRubi) [reply account_invoice_triple_discount/models/account_invoice.py:57]: @lmignon I don't know exactly why this works, I expected all the changes of #500 to work without any cache management.

The really wrong thing in my opinion is that the automated tests do not work without the `invalidate_cache` trick.
While debugging the tests issue (that note, 1 in 5 times were not failing!) I noticed the following (extracted from this same file):
```
    @api.multi
    @api.depends('discount2', 'discount3')
    def _compute_price(self):
        prev_values = self.triple_discount_preprocess()
        # Here discount is 50 and discount3 is 0, but inside the following method discount is 0 again.
        # More in particular, the record's discounts values change after a _browse call in the model's __iter__ method
        super(AccountInvoiceLine, self)._compute_price()
        self.triple_discount_postprocess(prev_values)
```
Moreover, this happened only for the 2nd of the lines created in `test_04`..
So I concluded that something was really messed up with the cache management

So I had to add the `invalidate_cache` line, but that somehow breaks the interface behavior as explained in https://github.com/OCA/account-invoicing/pull/517/commits/4411c63c9c2b807f8c17b8354e1ce85085a50f2a..

Comment (lmignon) [reply account_invoice_triple_discount/models/account_invoice.py:57]: @SimoRubi IMO it's not the right fix. If something is wrong with the cache we must fix it into Odoo. This error could  also reveal a problem into the way the 2 addons are designed.

Comment (SimoRubi) [reply account_invoice_triple_discount/models/account_invoice.py:57]: @lmignon I agree this is not the right fix, I would simply call this a workaround.
Anyway I probably won't have time to investigate further into the Odoo cache management and find the right fix (for these modules, or for the Odoo core).

**Comment by lmignon** [comment]
@SimoRubi Can you check #525. It fix the same issue in an other way and more details on what's wrong...

**Comment by SimoRubi** [comment]
@lmignon I have done my review in #525, please take a look over there to see the latest updates on these issues/PRs management

**Comment by SimoRubi** [comment]
@OCA/accounting-maintainers any news about this?
I understand from https://github.com/OCA/sale-workflow/issues/817#issuecomment-537047405 that this might be closed in the next future

**Comment by pedrobaeza** [comment]
Closing in favor of #525

**Pull request closed without merging**